packages/ingest/SftpProviderClient.js

```diff
@@ -67,17 +98,13 @@ class SftpProviderClient {
     return this.plaintextProviderKey;
   }
 
-  async getSftpClient() {
-    if (isNil(this.sftpClientInstance)) {
-      this.sftpClientInstance = new SftpClient({
-        host: this.host,
-        port: get(this.providerConfig, 'port', 22),
-        username: await this.getUsername(),
-        password: await this.getPassword(),
-        privateKey: await this.getPrivateKey()
-      });
+  /* @private */
+  getSftpClient() {
+    if (!this.connected) {
```

Comment:
What's the advantage of this pattern and the changes throughout here? Why throw an error in a method that's just returning the client if it's not connected?

Reply:
Because it forces you to explicitly connect before calling something like `download`?

Reply:
> Because it forces you to explicitly connect before calling something like `download`?

Correct. Since users need to explicitly end the connection, it is more clear when a connection needs to be closed if they are also explicitly connecting as well.